api/v1alpha1/nats_status.go

```diff
@@ -0,0 +1,87 @@
+package v1alpha1
+
+import (
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (ns *NatsStatus) IsEqual(status NatsStatus) bool {
+	thisWithoutCond := ns.DeepCopy()
+	statusWithoutCond := status.DeepCopy()
+
+	// remove conditions, so that we don't compare them
+	thisWithoutCond.Conditions = []metav1.Condition{}
+	statusWithoutCond.Conditions = []metav1.Condition{}
+
+	return reflect.DeepEqual(thisWithoutCond, statusWithoutCond) &&
+		ConditionsEquals(ns.Conditions, status.Conditions)
+}
+
+func (ns *NatsStatus) FindCondition(conditionType ConditionType) *metav1.Condition {
```

Comment:
Currently, this function is used inside the tests only. Is this the intended?

Reply:
`nats_status.go` file is not a test helper. The function is a helper with unit tests. Atm, its only used in tests.